corehq/apps/app_manager/suite_xml.py

```diff
@@ -2348,9 +2348,11 @@ def menus(self):
                 def get_commands():
                     for form in module.get_forms():
                         command = Command(id=id_strings.form_command(form))
-                        if module.all_forms_require_a_case() and \
-                                not module.put_in_root and \
-                                getattr(form, 'form_filter', None):
+                        if (
+                            (module.all_forms_require_a_case() or module.is_usercaseonly()) and
```

Comment:
Just changing this will cause breaking behaviour: 
* if the user references `#case` in the filter expression
* if the form is an `AdvancedForm`